pyanaconda/modules/payloads/source/cdrom/initialization.py

```diff
@@ -37,18 +39,66 @@ def name(self):
         return "Set up CD-ROM Installation Source"
 
     def _do_mount(self):
-        """Run CD-ROM installation source setup."""
-        log.debug("Trying to detect CD-ROM automatically")
+        """Run CD-ROM installation source setup.
 
+        Try to discover installation media and mount that. Device used for booting (inst.stage2)
+        have a priority.
+        """
+        log.debug("Trying to detect CD-ROM automatically")
         device_tree = STORAGE.get_proxy(DEVICE_TREE)
+
+        device_candidates = self._get_device_candidate_list(device_tree)
+        device_name = self._choose_installation_device(device_tree, device_candidates)
+
+        if not device_name:
+            raise SourceSetupError("Found no CD-ROM")
+
+        return device_name
+
+    def _get_device_candidate_list(self, device_tree):
+        stage2_device = self._probe_stage2_for_cdrom(device_tree)
+        device_candidates = device_tree.FindOpticalMedia()
+
+        if stage2_device:
+            device_candidates = [stage2_device] + device_candidates
+
+        return device_candidates
+
+    @staticmethod
+    def _probe_stage2_for_cdrom(device_tree):
+        if "repo" in kernel_arguments:
```

Comment:
I think we should document why "repo" in kernel arguments should skip the probing. I guess repo has priority as installation source from stage2.